server/controllers/addressController.ts

```diff
@@ -0,0 +1,97 @@
+import { Request, Response } from 'express'
+import { addressToLines, formatName } from '../utils/utils'
+import { AuditService, Page } from '../services/auditService'
+import { NameFormatStyle } from '../data/enums/nameFormatStyle'
+import logger from '../../logger'
+import AddressService from '../services/addressService'
+import GovSummaryItem from '../interfaces/GovSummaryItem'
+import Address from '../data/interfaces/prisonApi/Address'
+
+export default class AddressController {
+  constructor(
+    private readonly addressService: AddressService,
+    private readonly auditService: AuditService,
+  ) {}
+
+  public async displayAddresses(req: Request, res: Response) {
+    const { firstName, lastName, prisonerNumber, prisonId } = req.middleware.prisonerData
+    const { clientToken } = res.locals
+
+    const addresses = (await this.addressService.getAddresses(clientToken, prisonerNumber)) ?? []
+
+    const primaryAddressRecord = addresses.find(address => address.primary && !address.endDate)
+    const primaryAddress = this.mapAddressToSummaryItems(primaryAddressRecord)
+    const primaryAddressLabel = primaryAddressRecord?.mail ? 'Primary and mail address' : 'Primary address'
+
+    const mailAddress = this.mapAddressToSummaryItems(
+      addresses.find(address => address.mail && !address.primary && !address.endDate),
+    )
+
+    const otherAddresses = addresses
+      .filter(address => !address.primary && !address.mail && !address.endDate)
+      .sort((a, b) => b.startDate.localeCompare(a.startDate)) // Sort by ISO date string in reverse order
+      .map(this.mapAddressToSummaryItems)
+
+    this.auditService
+      .sendPageView({
+        userId: res.locals.user.username,
+        userCaseLoads: res.locals.user.caseLoads,
+        userRoles: res.locals.user.userRoles,
+        prisonerNumber,
+        prisonId,
+        correlationId: req.id,
+        page: Page.Addresses,
+      })
+      .catch(error => logger.error(error))
+
+    return res.render('pages/addresses/addresses', {
+      pageTitle: 'Addresses',
+      primaryAddressLabel,
+      primaryAddress,
+      mailAddress,
+      otherAddresses,
+      prisonerNumber,
+      prisonerName: formatName(firstName, '', lastName),
+      breadcrumbPrisonerName: formatName(firstName, '', lastName, { style: NameFormatStyle.lastCommaFirst }),
+    })
+  }
+
+  private mapAddressToSummaryItems(address: Address): { rows: GovSummaryItem[]; startDate: string } {
+    const addressSummary: GovSummaryItem[] = []
+
+    if (address) {
```

Comment:
If you initially include the guard `if(!address) return []` then you can proceed without the indentation. Can also avoid instantiating an empty array and pushing, can just define an array to return